### ISSUE TITLE
Editor bug and Switching from Editor to Viewer resets scroll position

### ISSUE THREAD

**Issue by gaganyadav80** [laurent22/joplin#5450]
## Environment

Joplin version: v2.4.6, AppImage
Platform: Linux
OS specifics: Opensuse Tumbleweed

## Issues I am having
The First issue is random and I don't know what triggered it. I was taking notes and joplin was in the background and when I switch back to it to edit some text in between a line as typed new characters it deleted the characters next to the cursor (watch the video carefully).

The second issue is if I am at a particular scroll position in the Editor and then the switch "Viewer" layout then the scroll position resets itself specially if I am near the end of the document then the "Viewer" layout goes all the way down where the scroll bar can go. Also you can see sometimes it glitches when switching to "Viewer" layout.

(Also, In Android, whatever the layout you switch to the scroll bar is reset to the very top of the document.)

## Steps to reproduce
Video for bug 1:

https://user-images.githubusercontent.com/47142137/132943463-0b405e1d-c1b7-4d06-aba5-43a3909166e0.mp4

Video for bug 2:

https://user-images.githubusercontent.com/47142137/132944070-bc404c5b-9878-4da6-92af-9c70535cd066.mp4

## Describe what you expected to happen
Bug 1: When editing inbetween a line then adding new character should not delete the characters already typed next to them

Bug 2: The scroll position can move a slight due to the difference in number of lines (markdown syntax and rendering difference) but should reset the scroll bar all the way to the end.

**Comment by rahul-gill** [comment]
Bug1 is not a bug, its a feature and every editor has it. You pressed the Insert key and switched to replace mode. Press the key again and you'll in insert mode, just the way you want.

**Comment by gaganyadav80** [comment]
You are right! Never knew there is a feature like that. Thanks for the info!
